### ISSUE TITLE
Weird issue after updating candle to 0.4.0

### ISSUE THREAD

**Issue by LaurentMazare** [huggingface/hf-hub#86]
I've tried to update candle to use the recently released 0.4.0 but am running into some weird issues that I haven't been able to debug yet. I'm not sure where the problem is coming from but just dropping a line here in case you can think of something.
The problem happens in the CI here: https://github.com/huggingface/candle/actions/runs/12549480881/job/34990753957?pr=2691
When running the same test locally, it indeed fails and the downloaded `model.safetensors` file for `bert-base-uncased` is different from the one I got with the previous version. However I wasn't able to make a standalone repro, e.g. the following code downloads the file correctly.
Would you have any thoughts on what might be causing this?

```
[laurent@lli test-hf-hub]$ cat src/main.rs 
use hf_hub::api::sync::Api;
fn main() {
    let api = Api::new().unwrap();
    let repo = api.model("bert-base-uncased".to_string());
    let weights = repo.get("model.safetensors").unwrap();
    let data = std::fs::read(&weights).unwrap();
    let digest = md5::compute(data);
    println!("{:x}  {:?}", digest, weights);
}
[laurent@lli test-hf-hub]$ cat Cargo.toml 
[package]
name = "test-hf-hub"
version = "0.1.0"
edition = "2021"

[dependencies]
hf-hub = { version = "0.4", features = ["tokio"] }
md5 = "0.7.0"
tokio = "1.29.1"
```

**Comment by Narsil** [comment]
Hmm both tests with timings that make both implementations race each other leading to weird errors since we added resumability.

Implementing a simple lockfile that should prevent that from happening (it might fail if the wait for lock is too long, I'm putting 30s for now).

**Comment by LaurentMazare** [comment]
Ah yeah that would definitely explaining. Having some form of lock sounds like a great idea in case a user launches multiple processes fetching the same file. Once this has been added, if you can make a new release that would be super helpful.

**Comment by Narsil** [comment]
Released 0.4.1 with locking.

**Comment by LaurentMazare** [comment]
Thanks!
